test/config/semian_config.rb

```diff
@@ -0,0 +1,29 @@
+require 'yaml'
+
+class SemianConfig
+  CONFIG_FILE = File.expand_path('../hosts.yml', __FILE__)
+
+  class << self
+    def [](service)
+      all.fetch(service)
+    end
+
+    def all
+      @entries ||= begin
+        entries = YAML.load_file(CONFIG_FILE)
+        flatten_keys(entries)
+      end
+    end
+
+    private
+
+    def flatten_keys(entries, object = {}, parent_key = '')
+      entries.each_with_object(object) do |(key, value), hash|
+        next flatten_keys(value, hash, key) if value.is_a?(Hash)
+
+        hash["#{parent_key}_#{key}"] = value
+        hash
+      end
+    end
+  end
+end
```

Comment:
@albiere I think it's still a bit over-complicated...

why not just a simple Hash... 

```
CONFIG_FILE = File.expand_path('../hosts.yml', __FILE__)
SemianConfig = YAML.load_file(CONFIG_FILE)
```

with the flattened config...

```
toxiproxy_upstream_host: 127.0.0.1
toxiproxy_upstream_port: 8474

mysql_host: 127.0.0.1
mysql_port: 3306
mysql_toxiproxy_port: 13306

redis_host: 127.0.0.1
redis_port: 6379
redis_toxiproxy_port: 16379

http_host: 127.0.0.1
http_port: 31050
http_toxiproxy_port: 31051
```